src/main/java/pkmncore/DBManager.java

```diff
@@ -0,0 +1,112 @@
+package pkmncore;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DBManager implements StorageUnit {
+
+    final private String dbURL;
+    final private String username;
+    final private String password;
+    final private String driver;
+
+    public DBManager(String dbURL, String username, String password, String driver) {
+        this.dbURL = dbURL;
+        this.username = username;
+        this.password = password;
+        this.driver = driver;
+    }
+
+    public Connection getConnection() throws PokemonError, SQLException {
+        setDriver();
+        return connectToDB();
+    }
+
+    public void save(String name, String height, String[] abilities) throws PokemonError {
+        savePokemonDetails(name, height, abilities);
+    }
+
+    public List<List<String>> getPokemon() throws PokemonError {
+        return getAllPokemon();
+    }
+
+    private void savePokemonDetails(String name, String height, String[] abilities) throws PokemonError {
+        try {
+            Connection connection = getConnection();
+            Statement statement = connection.createStatement();
+            saveNameAndHeight(name, height, statement);
+            saveAbilites(name, abilities, statement);
+            closeConnections(connection, statement);
+        } catch (SQLException e) {
+            throw new PokemonError(name + " has already been caught!");
+        }
+    }
+
+    private List<List<String>> getAllPokemon() throws PokemonError {
+        try {
+            Connection connection = getConnection();
+            Statement statement = connection.createStatement();
+            String sql = "SELECT * FROM POKEMON;";
+            List<List<String>> pokemon = listPokemon(connection, statement.executeQuery(sql));
+            closeConnections(connection, statement);
+            return pokemon;
+        } catch (SQLException e) {
+            throw new PokemonError("Something is wrong with the database!");
+        }
+    }
+
+    private List<List<String>> listPokemon(Connection connection, ResultSet caughtPokemon) throws SQLException {
+        List<List<String>> allPokemon = new ArrayList<List<String>>();
+        while (caughtPokemon.next()) {
+            List<String> pokemon = new ArrayList<String>();
+            pokemon.add(caughtPokemon.getString("name"));
+            pokemon.add(caughtPokemon.getString("height"));
+            getAbilities(pokemon, connection);
+            allPokemon.add(pokemon);
+        }
+        return allPokemon;
+    }
+
+    private void getAbilities(List<String> pokemon, Connection connection) throws SQLException {
```

Comment:
Its a little weird to read `getAbilities` but to then see that it doesn't return anything.
Using one of the inputs (List<String> pokemon) as an implicit output (the method basically says "Hey, I will change that for you!") is a bit secretive...